examples/dtls-sock/Makefile

```diff
@@ -20,6 +20,7 @@ USEPKG += tinydtls
 
 # Pull in sock APIs
 USEMODULE += sock_dtls
+USEMODULE += sock_dtls_verify_public_key
```

Comment:
Does this do anything with the default `CONFIG_DTLS_PSK`?

Reply:
No it does not. I could be guarded away - on the other hand, the impact is very minor and can be seen as test case: psk still works even when ecc verification is active?

Reply:
Okay, summary:
- Even if `sock_dtls_verify_public_key` is used with `CONFIG_DTLS_PSK` the verification part is not being build until `ecc` is selected
- Verification should be the default anyway, also noted in leandros first comment to this pr "I'm wondering whether this feature should actually be opt-out instead."
- For user of this example it is easier to switch between `psk/ecc` rather than bothering them with activating verification
- It could get guarded in the makefile but the benefit is minimal with the downside of making the Makefile less approachable

We believe it can stay the way it is, @benpicco?